enterprise/server/workflow/service/service.go

```diff
@@ -268,6 +274,142 @@ func (ws *workflowService) GetWorkflows(ctx context.Context, req *wfpb.GetWorkfl
+		req.GetWorkflowId(),
+	).Take(wf).Error
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, status.NotFoundError("Workflow not found")
+		}
+		return nil, status.InternalError(err.Error())
+	}
+
+	// Authorize workflow access
+	wfACL := perms.ToACLProto(&uidpb.UserId{Id: wf.GroupID}, wf.GroupID, wf.Perms)
+	if err := perms.AuthorizeRead(&user, wfACL); err != nil {
+		return nil, err
+	}
+
+	// Execute
+	isRepoPrivate := false
+	// TODO: Support other Git providers. We default to false on those for now,
+	// which means that we don't pass secrets even for private repos.
+	if isGitHubURL(wf.RepoURL) {
+		isRepoPrivate, err = github.IsRepoPrivate(ctx, wf.AccessToken, wf.RepoURL)
+		if err != nil {
+			return nil, status.WrapErrorf(err, "failed to determine whether repo is private")
+		}
+	}
+	// TODO: Refactor to avoid using this WebhookData struct in the case of manual
+	// workflow execution, since there are no webhooks involved when executing a
+	// workflow manually.
+	wd := &webhook_data.WebhookData{
+		PushedBranch:  req.GetBranch(),
+		TargetBranch:  req.GetBranch(),
+		RepoURL:       wf.RepoURL,
+		SHA:           req.GetCommitSha(),
+		IsRepoPrivate: isRepoPrivate,
+	}
+	invocationUUID, err := guuid.NewRandom()
+	if err != nil {
+		return nil, err
+	}
+	invocationID := invocationUUID.String()
+	extraCIRunnerArgs := []string{
+		fmt.Sprintf("--action_name=%s", req.GetActionName()),
+		fmt.Sprintf("--invocation_id=%s", invocationID),
+	}
+
+	executionID, err := ws.executeWorkflow(ctx, wf, wd, extraCIRunnerArgs)
+
+	if err := ws.waitForWorkflowInvocationCreated(ctx, executionID, invocationID); err != nil {
+		return nil, err
+	}
+
+	return &wfpb.ExecuteWorkflowResponse{InvocationId: invocationID}, nil
+}
+
+func (ws *workflowService) waitForWorkflowInvocationCreated(ctx context.Context, executionID, invocationID string) error {
+	executionClient := ws.env.GetRemoteExecutionClient()
+	indb := ws.env.GetInvocationDB()
+
+	errCh := make(chan error)
+	opCh := make(chan *longrunning.Operation)
+
+	waitStream, err := executionClient.WaitExecution(ctx, &repb.WaitExecutionRequest{
+		Name: executionID,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Listen on operation stream in the background
+	go func() {
+		for {
+			op, err := waitStream.Recv()
+			if err == io.EOF {
+				return
+			}
+			if err != nil {
+				errCh <- err
+				return
+			}
+			opCh <- op
+		}
+	}()
+
+	stage := repb.ExecutionStage_UNKNOWN
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case err := <-errCh:
+			return err
+		case op := <-opCh:
+			stage = operation.ExtractStage(op)
+		case <-time.After(1 * time.Second):
+			// Fall through
+		}
+		if stage == repb.ExecutionStage_EXECUTING || stage == repb.ExecutionStage_COMPLETED {
+			_, err := indb.LookupInvocation(ctx, invocationID)
+			if err == nil {
+				return nil
+			}
+			if !db.IsRecordNotFound(err) {
+				return err
+			}
```

Comment:
kinda cheeky, but could to shorten this to:

if _, err := indb.LookupInvocation(ctx, invocationID); !db.IsRecordNotFound(err) {
  return err
}

Reply:
I prefer the explicit nil check + return nil because otherwise the reader has to do a bit of extra thinking to convince themselves that this function can ever return `nil`.